x/executionlayer/genesis.go

```diff
@@ -40,8 +41,19 @@ func InitGenesis(
 		keeper.SetGenesisAccounts(ctx, data.Accounts)
 	}
 	keeper.SetChainName(ctx, data.ChainName)
+	stateHash, bonds, errStr := grpc.Commit(keeper.client, util.DecodeHexString(util.StrEmptyStateHash), response.GetSuccess().GetEffect().GetTransformMap(), genesisConfig.GetProtocolVersion())
```

Comment:
How about locating the Commit call upfront of keeper.SetXXX and remove temporal variable candidateBlock?
like
```go
	if reflect.TypeOf(response.GetResult()) != reflect.TypeOf(&ipc.GenesisResponse_Success{}) {
		panic(response.GetResult())
	}

	stateHash, bonds, errStr := grpc.Commit(keeper.client, util.DecodeHexString(util.StrEmptyStateHash), response.GetSuccess().GetEffect().GetTransformMap(), genesisConfig.GetProtocolVersion())
	if errStr != "" {
		panic(errStr)
	}

	if data.Accounts != nil {
		keeper.SetGenesisAccounts(ctx, data.Accounts)
	}
	keeper.SetChainName(ctx, data.ChainName)
	keeper.SetGenesisConf(ctx, data.GenesisConf)
	keeper.SetCandidateBlock(ctx, types.CandidateBlock{
		Hash:  []byte(types.GenesisBlockHashKey),
		Bonds: bonds,
	})
	keeper.SetEEState(ctx, []byte(types.GenesisBlockHashKey), stateHash)
```

Reply:
👍 
I update to this same way